File: TrappingRainWater/javascript/solution.test.js
var assert = require("assert");
var { trap } = require("./solution.js");

describe('Trapping rain water works for', () => {
    it('empty', () => { assert.equal(trap([]), 0); });
    it('hill', () => { assert.equal(trap([0, 1, 2, 1, 0]), 0); });
    it('bowl', () => { assert.equal(trap([2, 1, 0, 1, 2]), 4); });
    it('normal', () => { assert.equal(trap([0, 1, 0, 2, 1, 0, 1, 3, 2, 1, 2, 1]), 6); });
});

File: TrappingRainWater/elixir/solution_test.exs
Code.require_file("./solution.exs")

ExUnit.start

defmodule TrappingRainWaterTest do
  use ExUnit.Case

  test "empty" do
    assert 0 == TrappingRainWater.trap([])
  end

  test "hill" do
    assert 0 == TrappingRainWater.trap([0, 1, 2, 1, 0])
  end

  test "bowl" do
    assert 4 == TrappingRainWater.trap([2, 1, 0, 1, 2])
  end

  test "normal" do
    assert 6 == TrappingRainWater.trap([0, 1, 0, 2, 1, 0, 1, 3, 2, 1, 2, 1])
  end
end
